gtclang/CMakeLists.txt

```diff
@@ -67,7 +67,7 @@ if(GTCLANG_HAS_CUDA)
 endif(GTCLANG_HAS_CUDA)
 
 # Set C++ standard
-set(CMAKE_CXX_STANDARD 17)
+yoda_set_cxx_standard(c++17)

```

Comment:
Not 100% sure, but this might break cuda compilation.

Reply:
For some reason, CMake wasn't adding `-std=c++17` when I was compiling with clang-6 with `set(CMAKE_CXX_STANDARD 17)`, but this changed (for some reason) fixed that. Can we tell from the jenkins tests whether that breaks CUDA builds?

Reply:
yes, jenkins will test the cuda back end. so if everything passes, you are fine

Reply:
It breaks CUDA compilation because nvcc doesn't support c++17, therefore generated CUDA code must be c++11 compatible (or actually c++14 compatible, but today it is c++11) and `-std=c++17` must not be passed to the compile command. `yoda_set_cxx_standard()` is not really modern CMake and sets it globally for all targets. Adding it only to CMAKE_CXX_STANDARD will not apply it to CUDA files.
I am very surprised that `set(CMAKE_CXX_STANDARD 17)` doesn't work for MacOS. Where is the flag missing? Everywhere or only for some files?

Reply:
Ah yes, I remember now that CUDA currently only supports c++14. I'll try this again on mac soon and report back. Perhaps this was a symptom of another problem I already fixed elsewhere.

Reply:
This is not necessary after all.